bpf/tests/pktgen.h

```diff
@@ -0,0 +1,507 @@
+/* SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause) */
+/* Copyright Authors of Cilium */
+
+#ifndef __TEST_PKTGEN__
+#define __TEST_PKTGEN__
+
+#include <bpf/compiler.h>
+#include <bpf/builtins.h>
+#include <bpf/helpers.h>
+
+#include <lib/endian.h>
+
+#include <linux/byteorder.h>
+#include <linux/ip.h>
+#include <linux/ipv6.h>
+#include <linux/in.h>
+#include <linux/if_ether.h>
+#include <linux/tcp.h>
+
+/* A collection of pre-defined Ethernet MAC addresses, so tests can reuse them
+ *  without having to come up with custom addresses.
+ */
+
+#define mac_one   {0xDE, 0xAD, 0xBE, 0xEF, 0xDE, 0xEF}
+#define mac_two   {0x13, 0x37, 0x13, 0x37, 0x13, 0x37}
+#define mac_three {0x31, 0x41, 0x59, 0x26, 0x35, 0x89}
+#define mac_four  {0x0D, 0x1D, 0x22, 0x59, 0xA9, 0xC2}
+#define mac_five  {0x15, 0x21, 0x39, 0x45, 0x4D, 0x5D}
+#define mac_six   {0x08, 0x14, 0x1C, 0x32, 0x52, 0x7E}
+
+/* A collection of pre-defined IP addresses, so tests can reuse them without
+ *  having to come up with custom ips.
+ */
+
+#define IPv4(a, b, c, d) (((d) << 24) + ((c) << 16) + ((b) << 8) + (a))
```

Comment:
Why this and not the use of something like `inet_pton` and the standard in*_addr structs?

Reply:
We can't use `inet_pton`, it is not available in eBPF, even if it was, it takes addresses as `char*` so I am not convinced clang would be able to inline that function if we were to for example copy it. As for the `in*_addr` structs, this might be useful when we add IPv6, but the IPv4 version (`in_addr`) just [contains](https://github.com/openbsd/src/blob/master/include/arpa/inet.h#L110) a single 32-bit field called `s_addr` so I don't feel like it adds much.